src/app/shared/services/snack.service.ts

```diff
@@ -41,4 +41,9 @@ export class SnackService {
       duration: 10000,
     });
   }
+  public showSendDownlinkFailNoDevices(){
```

Comment:
This is specific to downlink. It's not snack service's responsibility to know about it.
Move it to downlink service or something more suitable